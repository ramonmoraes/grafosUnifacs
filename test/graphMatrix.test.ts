import Graph from "../src/graph";
import GraphNode from "../src/graphNode";
import GraphMatrix from '../src/graphMatrix';


describe("Marix", () => {
  let graph: Graph;
  let ismeloNode: GraphNode;
  let dartNode: GraphNode;

  beforeEach(() => {
    ismeloNode = new GraphNode({
      identifier: "ismelo",
      attributes: { lang: ["en", "pt"] }
    });

    dartNode = new GraphNode({
      identifier: "dart",
      attributes: { lang: ["en"] }
    });

    graph = new Graph([ismeloNode, dartNode]);
    graph.createLinks();
  });

  test("get adjacentGraphMatrix", () => {
    const sample = new GraphMatrix(graph).adjacentGraphMatrix({filteredValue: "en"})
    expect(sample).toMatchSnapshot();
  });

  test("adjacentGraphMatrix conversion to Dijkstra", () => {
    const sample = new GraphMatrix(graph).adjacentGraphMatrix({filteredValue: "en"})
    expect(sample).toMatchSnapshot();
  });
});

