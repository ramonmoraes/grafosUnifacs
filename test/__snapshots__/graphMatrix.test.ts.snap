// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marix get adjacentGraphMatrix 1`] = `
Array [
  Array [
    0,
    1,
  ],
  Array [
    1,
    0,
  ],
]
`;
